### PR TITLE
Update robot hardware, update docs, refactor and add examples

### PR DESCRIPTION
**Pull request by nqyy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -406,15 +401,39 @@ For example, [object_pose_estimator_3d](mbodied/agents/sense/object_pose_estimat
 
 The [motion_controls](mbodied/types/motion_controls.py) module defines various motions to control a robot as Pydantic models. They are also subclassed from `Sample`, thus possessing all the capability of `Sample` as mentioned above. These controls cover a range of actions, from simple joint movements to complex poses and full robot control.
 
-### Hardware Interface
+### Robot
 
-Mapping robot actions from a model to an action is very easy. In our example script, we use a mock hardware interface. We also have an [XArm interface](mbodied/hardware/xarm_interface.py) as an example.
+You can integrate your custom robot hardware by subclassing [RobotBase](mbodied/robot/robot_base.py) quite easily. You only need to implement `do()` function to perform actions (and some additional methods if you want to record dataset on the robot). In our examples, we use a [mock robot](mbodied/robot/sim_robot.py). We also have an [XArm robot](mbodied/robot/xarm_robot.py) as an example.
```
[the remark this answers is not in the document]
Comment (nqyy) [reply README.md:406]: Done.

File: README.md [hunk 2]
```diff
@@ -406,15 +401,39 @@ For example, [object_pose_estimator_3d](mbodied/agents/sense/object_pose_estimat
...
+
+robot = SimRobot()
+robot_recorder = RobotRecorder(robot, record_frequency=5)
+
+robot_recorder.start_recording(task="pick up the fork")
+motion = HandControl.unflatten([0.1, 0.2, 0.0, 0.0, 0.0, 0.0, 1.0])
+robot.do(motion)
+robot.do(motion)
+robot_recorder.stop_recording()
+```
```
[the remark this answers is not in the document]
Comment (nqyy) [reply README.md:425]: Done.

**Comment by nqyy** [comment]
> Lets put simple robot agent colab in the readme maybe? Not seeing it linked anywhere:
> 
> https://colab.research.google.com/drive/10Tbrg5u1GyDOHCXOkd5xOPRX2-o-E1vb?usp=sharing

We have a next PR with all the Colabs rewritten.

**Comment by sebbyjp** [comment]
Lets move this colab in this PR. Also robot.py not robot_base.py still confusing

**Review by sebbyjp — APPROVED** [review]

**Review by Zaid-Waghoo — APPROVED** [review]

**Pull request merged**